qfieldsync/core/cloud_api.py

```diff
@@ -244,22 +358,33 @@ def server_url(self):
 
         return re.sub(r"([^:]/)(/)+", r"\1", url)
 
+    def get_auth_capabilities(self) -> QNetworkReply:
+        """Get authentication capabilities from a server"""
+        reply = self.cloud_get("auth/providers/")
+        return reply
+
     def auto_login_attempt(self):
-        cfg = self.auth()
+        if self.auth_method == CloudAuthMethod.CREDENTIALS:
+            cfg = self.auth()
+
+            server_url = cfg.uri() or self.url
+            username = cfg.config("username")
+            password = cfg.config("password")
 
-        server_url = cfg.uri() or self.url
-        username = cfg.config("username")
-        password = cfg.config("password")
+            if username and password:
```

Comment:
This works, but "only by accident":
For `username = "foo"` and `password = "bar"`, the expression `username and password` will evaluate to `"bar"`.

So you're essentially doing `if "bar"` in that statement. Which works correctly, but is probably not what you intended.

A more idiomatic way in Python would be:
```python
if all([username, password]):
```
using the [`all()` builtin](https://docs.python.org/3/library/functions.html#all). That evaluates all the elements in the sequence to boolean, and returns `True` if all of them evaluate to `True`. If not, it will short-circuit and return `False` as soon as it encounters the first falsy value.